Makefile

```diff
@@ -821,6 +813,56 @@ ifneq ($(word 2,$(MAKECMDGOALS)),)
 export ENABLE_MSG_EXTRA := yes
 endif
 
+uavobjects_armsoftfp: TARGET=uavobjects_armsoftfp
+uavobjects_armsoftfp: OUTDIR=$(BUILD_DIR)/$(TARGET)
+uavobjects_armsoftfp: uavobjects
+	$(V1) mkdir -p $(OUTDIR)/dep
+	$(V1) cd $(ROOT_DIR)/flight/uavobjectlib && \
+		$(MAKE) -r --no-print-directory \
+		TCHAIN_PREFIX="$(ARM_SDK_PREFIX)" \
+		REMOVE_CMD="$(RM)" \
+		\
+		MAKE_INC_DIR=$(MAKE_INC_DIR) \
+		ROOT_DIR=$(ROOT_DIR) \
+		TARGET=$(TARGET) \
+		OUTDIR=$(OUTDIR) \
+		\
+		PIOS=$(PIOS) \
```

Comment:
Well, I plan to move the BUILD_DIR one.   You made a good point that it makes sense to make sure that things are contained.

OTOH, the makefile stuff is rotting.  Most of the lines of target makefiles are dead and have no purpose.  They've been unmaintained.  They *have* been carelessly copy and pasted for a long time.  I really would like to refactor and clean the makefiles in future PRs, but I don't want to do it this way.  Instead, I'd like to export a few more paths and let common included files calculate a lot more.